Update Haskell package set to ghc-8.10.3 (plus other fixes)

peti opened the pull request:
This PR is test-built by Hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. I'll fix up the remaining errors and merge it on Friday, 2020-12-25 20:00 +01:00. You can watch this live on Twitch at https://www.twitch.tv/peti343. In addition to the chat features offered by Twitch, there is also a voice conference at https://discord.gg/YTEa3XR that viewers can use to chat with me and with each other.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-36-ge221522 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/7c122b0d2de747b18ecc3cfb1318f9eb83301f81.

clkamp committed:
haskellPackages.taffybar: Mark as unbroken

sternenseemann committed:
haskellPackages.gitit: jailbreak

Should fix the build failure described in #107209 for master/unstable.

sternenseemann committed:
haskellPackages.flat: unbreak

0.4.4 fixed the build

pbogdan committed:
haskellPackages.greenclip: don't mark as broken

sternenseemann committed:
haskellPackages.apecs-physics: depend on ApplicationServices on darwin

cdepillabout committed:
spago: 0.17.0 -> 0.18.0

peti committed:
all-cabal-hashes: update to Hackage at 2020-12-24T01:16:25Z

sternenseemann commented:
Would be nice to get `apecs-physics` 0.4.5 in with this update. I don't know whether stackage nightly will update to it, but the update unbreaks `apecs-physics`, `apecs-gloss` and `apecs-physics-gloss`. See <https://github.com/NixOS/nixpkgs/pull/107395#issuecomment-750223947>.

sternenseemann committed:
haskellPackages.apecs-physics: push to 0.4.5 to unbreak build

apecs-physics was broken by the update of glibc to 2.32 due to the
vendored Chipmunk2D using sys/sysctl.h on GNU/Linux. 0.4.5 updates the
vendored version of Chipmunk2D which resolves this issue.

Resolves #107358.

peti committed:
haskellPackages: update default compiler from ghc 8.10.2 to version 8.10.3

peti committed:
haskell-ListLike: patch to fix the build with ghc-8.10.3

peti committed:
hackage2nix: update list of broken builds to fix evaluation on Hydra

peti commented:
We'll merge https://github.com/NixOS/nixpkgs/pull/107593 next week.
